pkg/collector/corechecks/system/disk/disk/disk.go

```diff
@@ -148,39 +268,428 @@ func (c *Check) instanceConfigure(data integration.Data) error {
+				log.Debugf("Error decoding: %s\n", err)
+				continue
+			}
+			if len(m.Host) == 0 || len(m.Share) == 0 {
+				log.Errorf("Invalid configuration. Drive mount requires remote machine and share point")
+				continue
+			}
+			log.Debugf("Mounting: %s\n", m)
+			mountType, remoteName := m.RemotePath()
+			log.Debugf("mountType: %s\n", mountType)
+			err = NetAddConnection(mountType, m.MountPoint, remoteName, m.Password, m.User)
+			if err != nil {
+				log.Errorf("Failed to mount %s on %s: %s", m.MountPoint, remoteName, err)
+				continue
+			}
+			log.Debugf("Successfully mounted %s as %s\n", m.MountPoint, remoteName)
+		}
+	}
 	return nil
 }
 
-func stringSliceContain(slice []string, x string) bool {
-	for _, e := range slice {
-		if e == x {
-			return true
+func (c *Check) configureExcludeDevice(instanceConfig map[interface{}]interface{}, initConfig map[interface{}]interface{}) error {
+	for _, key := range []string{"device_global_exclude", "device_global_blacklist"} {
+		if deviceExclude, ok := initConfig[key].([]interface{}); ok {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+				}
+			}
 		}
 	}
-	return false
+	for _, key := range []string{"device_exclude", "device_blacklist", "excluded_disks"} {
+		if deviceExclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range deviceExclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+				}
+			}
+		}
+	}
+	excludedDiskRe, found := instanceConfig["excluded_disk_re"]
+	if excludedDiskRe, ok := excludedDiskRe.(string); found && ok {
+		var err error
+		regexp, err := regexp.Compile(excludedDiskRe)
+		if err != nil {
+			return err
+		}
+		c.cfg.excludedDevices = append(c.cfg.excludedDevices, *regexp)
+	}
+	return nil
 }
 
-func (c *Check) applyDeviceTags(device, mountpoint string, tags []string) []string {
-	// apply device/mountpoint specific tags
-	for re, deviceTags := range c.cfg.deviceTagRe {
-		if re == nil {
+func (c *Check) configureIncludeDevice(instanceConfig map[interface{}]interface{}) error {
+	for _, key := range []string{"device_include", "device_whitelist"} {
+		if deviceInclude, ok := instanceConfig[key].([]interface{}); ok {
+			for _, val := range deviceInclude {
+				if strVal, ok := val.(string); ok {
+					regexp, err := regexp.Compile(strVal)
+					if err != nil {
+						return err
+					}
+					c.cfg.includedDevices = append(c.cfg.includedDevices, *regexp)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (c *Check) configureExcludeFileSystem(instanceConfig map[interface{}]interface{}, initConfig map[interface{}]interface{}) error {
+	defaultFileSystems := []interface{}{"iso9660$", "tracefs$"}
```

Comment:
I would add a test that exercise that default values are respected when not configured them manually, tha way in the future we ensure no one removes them from here. 

I would add the test for these values and for `defaultMountPoints`

Also, out of curiosity, have you consider configuring the default values instead of here in the `newDiskConfig` function. It might be more obvious to the reader where the default values are set.

Reply:
Good catch! Done